docs/gitops-generation.md

```diff
@@ -0,0 +1,7 @@
+# GitOps Generation
+
+The generation of GitOps resources is handled by the go module [redhat-developer/gitops-generator](https://github.com/redhat-developer/gitops-generator). Please refer to the module's README for more information on the usage.
+
+### Development
+
+When working on a story that requires contribution to [redhat-developer/gitops-generator](https://github.com/redhat-developer/gitops-generator), replace the gitops-generator go module's path in your local `go.mod` with a gitops-generator repository's local path to ensure application-service is working as expected.
```

Comment:
Do you want to add an example? e.g.
`replace github.com/redhat-developer/gitops-generator => <path-to-local-gitops-generator-repo>`

Since the changes will need to be coordinated once the story is complete, we should also add instructions to ensure: 
1.  Gitops-generator changes are committed and merged first
2. Run  `go list -m -json github.com/redhat-developer/gitops-generator@latest` to determine the latest version
3. Update the application-service `go.mod` to 1) remove the replace statement above and 2) update the version in the `github.com/redhat-developer/gitops-generator` dependency

Reply:
@kim-tsao  Updated! Feel free to review https://github.com/redhat-developer/gitops-generator/pull/42 as well!